ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -390,16 +390,6 @@ public void commit (boolean report_heuristics) throws HeuristicMixed,
 			}
 
 			destroyAction();
-		} else if (outcome == ActionStatus.ABORTED || outcome == ActionStatus.H_ROLLBACK) {
-			throw new TRANSACTION_ROLLEDBACK(ExceptionCodes.FAILED_TO_COMMIT,
```

Comment:
You do not need to change anything in the commit method since we are already checking for the ABORTED state and we correctly throw the TRANSACTION_ROLLEDBACK exception.